### PR TITLE
msvc build [WIP]

### PR DESCRIPTION
**Pull request by justnope**
This is a work in progress but I'm able to build & use the dlls it produces.
The open issues are:
* ~~svnversion.h isn't generated~~ (only git is supported for version info in ms, svn can be added if needed)
* ~~version check of portaudio needs to be put back in~~
* ~~msvc 2017 32bit fails, but 2015 32bit works~~
* check which WIN32 usage should be replaced by _MSC_VER
...

To build:
$env:Path= "\<pythonpath\>;" + $env:Path
$env:Path= "\<jack2srcpath\>/windows;" + $env:Path
windows/install_vcpkg_deps.bat
$env:INCLUDE="<path>/windows/deps/include"
$env:LIB="<path>/windows/deps/lib_x64"
python waf configure --debug --msvc_targets="x64" --doxygen=no
python waf build

### CODE REVIEW

**Comment by 7890** [comment]
Thanks for the update. Would it be possible for you to attach a full verbose build log here? There is currently no MSVC CI so it can only be tested by developers with MSVC installed.

The builds on Linux and OSX look good according to travis. I'll try building this branch with MXE and see if we can unify changes needed to potentially handle issues for both windows build chains at once. It's good to have the WIP PR here so any changes will go through the normal CI but please don't expect a quick merge.

It needs a closer look by more eyes, changing library exports etc. must be reviewed for possible compatibility issues etc. Theoretically speaking, a newly built client .exe program should work with an existing JACK install, also using newly built JACK libraries and server should work along with existing client .exe programs. There are a lot of details to consider for the binaries to work as expected in deployed system so having a going-through build is just the start. We can make it happen sometime, thanks for your support so far.

**Comment by justnope** [comment]
> Thanks for the update. Would it be possible for you to attach a full verbose build log here? There is currently no MSVC CI so it can only be tested by developers with MSVC installed.

I've added appveyor to my repo:
https://ci.appveyor.com/project/justnope/jack2

But with my latest commit I seem to have broken the clang build, but I don't know why.

> It's good to have the WIP PR here so any changes will go through the normal CI but please don't expect a quick merge.

np, actually I thought it would take several phases. This whole pr is a proof of concept and hopefully we can cherry-pick the less controversial commits to be included in master. This is to avoid having to rebase things.

I'm actually curious about your work with MXE and mingw. If you have something to share, I'd like to try it out.

**Comment by 7890** [comment]
Note: PR replaces https://github.com/jackaudio/jack2/pull/318

File: wscript [hunk 1]
```diff
@@ -91,31 +91,31 @@ def options(opt):
...
-            args='--cflags --libs')
+            package='libffado',
+            args=['--cflags', '--libs', 'libffado >= 1.999.17'])
     freebob = opt.add_auto_option(
             'freebob',
             help='Enable FreeBob driver')
     freebob.check_cfg(
-            package='libfreebob >= 1.0.0',
-            args='--cflags --libs')
+            package='libfreebob',
```
Comment (7890) [inline wscript]: There is no freebob anymore. It was completely replaced by firewire / FFADO

Comment (justnope) [reply wscript]: I'm not sure what to do.
Do you want me to fix it in this branch or issue a separate pr so it's fixed in master?

Comment (7890) [reply wscript]: It was just a hint that the current wscript in the master branch has removed all code related to Freebob.

Comment (justnope) [reply wscript]: oh gotcha. I've rebased the code.

**Comment by Schroedingers-Cat** [comment]
Is there a reason this PR got closed without being merged?

**Comment by 7890** [comment]
I guess it was WIP but only @justnope can tell why it was closed. It looks like a lot of work was put into this.

**Comment by justnope** [comment]
First, I want to stress that I don't believe there's any ill will from any of the jack developer and certainly not from me.

I closed it because I recognized a pattern. 
* Windows isn't a priority so PRs with windows compat. updates gets delayed
* Jack2 keeps getting developed
* The PR can't be merged anymore so a request to rebase it is made
* The PR is rebased after awhile: now goto beginning.

In one of my replies I told you I wasn't expecting to get this entire PR committed but instead parts of it. So when there's an rebase, I'll have a lot less work.

**Pull request closed without merging**